src/components/suggest/suggest.tsx

```diff
@@ -60,7 +60,14 @@ export interface EuiSuggestionProps
 }
 
 type _EuiSuggestProps = CommonProps &
-  EuiFieldSearchProps & {
+  Omit<
+    EuiFieldSearchProps,
+    | 'isLoading' // Use status.loading instead for consistency
+    | 'onChange' // Use onSearchChange instead for consistency
+    | 'onSearch' // Use onSearchChange instead for consistency
+    | 'incremental' // Hard-coded by EuiSelectableSearch, and must be true because EuiSuggest doesn't respond to Enter keypresses
+    | 'inputRef' // Hard-coded by EuiSelectable - not supported as a consumer-passed prop, although the input element can be obtained via onInputChange
+  > & {
```

Comment:
Not totally sure if these are considered breaking changes - but I removed them because they felt confusing and unnecessary (especially in the playground/props documentation), and I don't think we want users passing these props for the commented reasons

Reply:
Or at the very least, why are we extending `EuiFieldSearchProps` here instead of `EuiSelectableSearchProps`?
```suggestion
  Omit<
    EuiSelectableSearchProps,
    | 'isLoading' // Use status.loading instead for consistency
    | 'inputRef' // Hard-coded by EuiSelectable - not supported as a consumer-passed prop, although the input element can be obtained via onInputChange
  > & {
```

Reply:
> why are we extending EuiFieldSearchProps here instead of EuiSelectableSearchProps?

We can't just extend `EuiSelectableSearchProps` because it has props passed from `EuiSelectable` used by its internal logic that **aren't** passed down to the underlying `EuiFieldSearch`, and also hard-coded props (like `inputRef` that other components cannot manually pass). At the most we could export a shared `EuiSelectableFieldSearch` props of omitted EuiFieldSearchProps for all components that use EuiSelectable, but as there's only 2 major components currently that use 

> To be honest, I don't think we should have established a specific onSearchChange in the previous PR, but relied on the same props that exist on EuiFieldSearch (onSearch and onChange):

I don't disagree and I'd be fine making that breaking change if @thompsongl is. I will say however that I think it's perfectly valid to not blanket accept all props that either `EuiFieldSearch` takes or `EuiSuggest.searchProps` takes just because we can. Certain components that re-use `EuiSelectable` may not want to make every prop of `EuiSelectable`/`EuiFieldSearch` customizable, and I think that's a completely valid choice. For example, the `onSearch` and `incremental` prop of EuiFieldSearch makes 0 sense as far as I can tell in almost all usages of EuiSelectable as the enter key is not registered whatsoever, so we should consider eliminating the `onSearch` prop in favor of `onChange` (which would be yet another breaking change).

Reply:
Makes sense to me 👍

Reply:
Alrighty, here's what I pushed up to address this feedback:

- Per Caroline's 2nd comment, I DRY'd out `EuiSelectableSearchProps` so that `EuiSuggest` now reuses it instead of `EuiFieldSearchProps` (https://github.com/elastic/eui/pull/5658/commits/e36162f01c36248a40927ab50a0118ceb07b12d5)
- Per Caroline's 1st comment and Greg's 1st comment, I renamed `EuiSuggest`'s `onInputChange` to just `onInput` and `onSearchChange` to `onSearch` (https://github.com/elastic/eui/pull/5658/commits/d37d0f4d9528af1188b787d4191815f7906f212e)
  -  Feedback appreciated on this one, my rationale for using `onInput` and not `onChange` was because of the semi-confusing arguments. onChange callbacks typically return an event, whereas this callback returns **just** the input DOM node and not the full event., so I figured `onInput` made more sense (and is more 1:1 with what we're passing directly to the underlying component(s)).
- Per my last comment, I deprecated `EuiSelectable`'s `searchProps.onSearch` in favor of `searchProps.onChange` (https://github.com/elastic/eui/pull/5658/commits/ee36eda1ee4f340c67f6d2527030eeb9b6c071ae)

Reply:
^ Thanks for these